### PR TITLE
Fixes get_exact_dist() and Improves Reagent Checking

### PR DESCRIPTION
**Pull request by Shadowmech88**
Fixes if() arguments in get_exact_dist().
Adds new procs for reagent checking, to cut down on clutter.

### CODE REVIEW

**Comment by Aurx** [comment]
Unatomic
-MAX_INT/10 
DNM
CONTROVERSIAL
DECONTRIBUTOR HE

**Comment by Shadowmech88** [comment]
I asked if it would be a problem in the Discord and no one seemed to care, so I figured why not.

File: code/modules/reagents/Chemistry-Holder.dm [hunk 1]
```diff
@@ -608,6 +608,19 @@ trans_to_atmos(var/datum/gas_mixture/target, var/amount=1, var/multiplier=1, var
 		return amount_cache[reagent] >= max(0,amount)
 	return 0
 
+/datum/reagents/proc/has_any_reagents(var/list/input_reagents, var/amount = -1)		//returns true if any of the input reagents are found
+	var/has = FALSE
```
Comment (PJB3005) [inline code/modules/reagents/Chemistry-Holder.dm]: You can just use the `.` variable which is built in, so you can remove the `return has`

File: code/modules/reagents/Chemistry-Holder.dm [hunk 2]
```diff
@@ -608,6 +608,19 @@ trans_to_atmos(var/datum/gas_mixture/target, var/amount=1, var/multiplier=1, var
...
+	for(var/i in input_reagents)
+		if(has_reagent(i, amount))
+			has = TRUE
```
Comment (PJB3005) [inline code/modules/reagents/Chemistry-Holder.dm]: Make this `return TRUE`

**Review by PJB3005 — CHANGES_REQUESTED** [review]

**Review by PJB3005 — APPROVED** [review]

**Pull request merged**